### DISCUSSION TITLE
Make `FFinstance instance` a global variable

### DISCUSSION THREAD

**Discussion by CarterLi** [fastfetch-cli/fastfetch#473] [General]
We pass `instance` anywhere, and half of them are `FF_UNUSED`.

Thoughts? @LinusDierheimer

**Comment by LinusDierheimer** [comment]
Currently it makes sense, yes. But if we are going to make a library one day, i think having an instance variable is a better design.

**Reply by CarterLi** [reply]
> if we are going to make a library one day, i think having an instance variable is a better design

If we make a library, we should better split `FFinstance` into smaller structures. It's way too bloated.
